src/Functions/ReplaceStringImpl.h

```diff
@@ -346,7 +369,14 @@ struct ReplaceStringImpl
         size_t input_rows_count)
     {
         if (needle.empty())
-            throw Exception(ErrorCodes::ARGUMENT_OUT_OF_BOUND, "Length of the pattern argument in function {} must be greater than 0.", name);
+        {
+            res_data.assign(haystack_data.begin(), haystack_data.end());
```

Comment:
Have you tested it?
Zero bytes also need to be set in `res_data`.

Reply:
Fix.